src/subscriptions/subscriptionReaper.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ */
+import { DynamoDbDataService, DynamoDb } from 'fhir-works-on-aws-persistence-ddb';
+
+const enableMultiTenancy = process.env.ENABLE_MULTI_TENANCY === 'true';
+const dbService = new DynamoDbDataService(DynamoDb, false, {
+    enableMultiTenancy,
+});
+
+const reaperHandler = async (event: any) => {
+    console.log('subscriptionReaper event', event);
+    try {
+        const subscriptions = await dbService.getActiveSubscriptions({});
+        const currentTime = new Date();
+        // filter out subscriptions without a defined end time.
+        // check if subscription is past its end date (ISO format)
+        // example format of subscriptions: https://www.hl7.org/fhir/subscription-example.json.html
+        return await Promise.all(
+            subscriptions
+                .filter((s: Record<string, any>) => s.end && currentTime >= new Date(s.end))
```

Comment:
I think we should consider the case where the end date is not in ISO format. Right now it will always be ignored as it is an `Invalid Date` that cannot be compared with currentTime.

I think this behavior it's ok, but maybe just add a log like "Skipping subscription {id} since end date is not in a valid format: {end.date}"

Reply:
I refactored the filter to log in the case that the condition fails due to no end date or an invalid end date